.github/workflows/rust_checks.yml

```diff
@@ -8,80 +8,63 @@ on:
 env:
   CARGO_TERM_COLOR: always
 
-jobs:
+jobs: 
   check:
     name: Minimal checks on non-zkvm crates
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Download and Extract Fixtures
-        run: |
-          chmod +x ./scripts/download-and-extract-fixtures.sh
-          ./scripts/download-and-extract-fixtures.sh
-
       - name: Install Rust toolchain
         uses: dtolnay/rust-toolchain@nightly
         with:
           components: clippy, rustfmt
-
-      - name: Cache Rust dependencies
-        uses: actions/cache@v4
+      
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v2
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
+          key: base 
 
       - name: Check formatting
         run: cargo fmt --check --all
 
       - name: Run tests
         run: cargo test -p zkevm-metrics -p witness-generator -p ere-hosts
 
+  download-eest-fixtures:
+    name: Download and Extract Fixtures
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Download and Extract Fixtures
+        run: |
+          chmod +x ./scripts/download-and-extract-fixtures.sh
+          ./scripts/download-and-extract-fixtures.sh
```

Comment:
We were downloading & extracting EEST fixtures many times with no good reason. Despite this was very fast, it was wasteful plus sometimes triggered a rate limiting since we also were doing it once per zkVM in the matrix below (and all at the ~same instant).